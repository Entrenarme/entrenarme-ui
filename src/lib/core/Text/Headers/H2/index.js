// @flow
import * as React from 'react';
import styled from 'styled-components';

import * as fonts from '../../../../helpers/fonts';
import colors from '../../../../helpers/colors';

const normalDesktop = 1.75;

const normalMobile = 1.25;

const sizeModifier = 0.25;

const textSizeDesktopTransform = props => {
  if (props.l) {
    return normalDesktop + sizeModifier;
  } else if (props.s) {
    return normalDesktop - sizeModifier;
  } else {
    return normalDesktop;
  }
};

const textSizeMobileTransform = props => {
  if (props.l) {
    return normalMobile + sizeModifier;
  } else if (props.s) {
    return normalMobile - sizeModifier;
  } else {
    return normalMobile;
  }
};

const StyledH2 = styled.h2`
  text-transform: uppercase;
  font-size: ${props => `${textSizeDesktopTransform(props)}rem`};
  font-weight: 600;
  font-family: ${fonts.mainFont};
  color: ${props => (props.color ? props.color : colors.primary.default)};

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    font-size: ${props => `${textSizeMobileTransform(props)}rem`};
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    font-size: ${props => `${textSizeMobileTransform(props)}rem`};
  }

  /* iPads (portrait and landscape) ----------- */
  @media only screen and (min-width: 768px) and (max-width: 1223px) {
    /* Styles */
  }

  /* Desktops and laptops ----------- */
  @media only screen and (min-width: 1224px) {
    /* Styles */
  }

  /* Large screens ----------- */
  @media only screen and (min-width: 1824px) {
    /* Styles */
  }

  /* ----------- Retina Screens ----------- */
  @media screen and (min-device-width: 1200px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
    /* Styles */
  }
`;

type Props = {
  children: React.Node,
  color?: string,
};

const H2 = ({ children, ...rest }: Props) => (
  <StyledH2 {...rest}>{children}</StyledH2>
);

H2.defaultProps = {};

export default H2;
