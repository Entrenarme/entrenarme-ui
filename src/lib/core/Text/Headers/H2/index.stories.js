// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import H2 from './index';

const ElementDecoration = storyFn => (
  <div style={{ paddingLeft: '15px' }}>{storyFn()}</div>
);

storiesOf('Text/Headers/H2', module)
  .addDecorator(ElementDecoration)
  .add('default', withInfo('Heading 2')(() => <H2>Heading 2</H2>));

storiesOf('Text/Headers/H2', module)
  .addDecorator(ElementDecoration)
  .add('small', withInfo('Heading 2')(() => <H2 s>Heading 2</H2>));

storiesOf('Text/Headers/H2', module)
  .addDecorator(ElementDecoration)
  .add('large', withInfo('Heading 2')(() => <H2 l>Heading 2</H2>));

storiesOf('Text/Headers/H2', module)
  .addDecorator(ElementDecoration)
  .add(
    'with color prop',
    withInfo('Heading 2')(() => <H2 color="red">Heading 2</H2>),
  );
