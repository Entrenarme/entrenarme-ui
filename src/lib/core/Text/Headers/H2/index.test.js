import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import H2 from './index';

test('renders "Heading" with no props', () => {
  const { container } = render(<H2>Heading</H2>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with large size', () => {
  const { container } = render(<H2 size="large">Heading</H2>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with small size', () => {
  const { container } = render(<H2 size="small">Heading</H2>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with color prop', () => {
  const { container } = render(<H2 color="warning">Heading</H2>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with text align', () => {
  const { container } = render(<H2 textAlign="center">Heading</H2>);
  expect(container.firstChild).toMatchSnapshot();
});
