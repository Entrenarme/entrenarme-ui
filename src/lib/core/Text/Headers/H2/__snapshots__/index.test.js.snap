// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Heading" with color prop 1`] = `
<h2
  class="sc-bdVaJa eGwGdc"
>
  Heading
</h2>
`;

exports[`renders "Heading" with large size 1`] = `
<h2
  class="sc-bdVaJa gJNZvU"
>
  Heading
</h2>
`;

exports[`renders "Heading" with no props 1`] = `
<h2
  class="sc-bdVaJa hcnaUU"
>
  Heading
</h2>
`;

exports[`renders "Heading" with small size 1`] = `
<h2
  class="sc-bdVaJa iYASYg"
>
  Heading
</h2>
`;

exports[`renders "Heading" with text align 1`] = `
<h2
  class="sc-bdVaJa jHlXIX"
>
  Heading
</h2>
`;
